index.js

```diff
@@ -535,6 +576,44 @@ methods.concat('all').forEach(function(method){
 })
 
 /**
+ * Find a path for the previously created named route. The name
+ * supplied should be separated by '.' if nested routing is
+ * used. Parameters should be supplied if the route includes any
+ * (e.g. {userid: 'user1'}).
+ *
+ * @param {string} route name or '.' separated path
+ * @param {Object} params
+ * @return {string}
+ */
+
+Router.prototype.findPath = function findPath(routePath, params) {
+  if (!((routePath instanceof String) || typeof routePath === 'string')) {
+    throw new Error('route path should be a string')
+  }
+  var firstDot = routePath.indexOf('.')
+  var routeToFind;
+  if (firstDot === -1) {
+    routeToFind = routePath
+  } else {
+    routeToFind = routePath.substring(0, firstDot)
+  }
+  var thisRoute = this.routes[routeToFind]
+  if (!thisRoute) {
+    throw new Error('route path \"'+ routeToFind + '\" does not match any named routes')
+  }
+  var toPath = pathToRegexp.compile(thisRoute.path)
+  var path = toPath(params)
+  if (firstDot === -1) { // only one segment or this is the last segment
+    return path
+  }
+  var subPath = routePath.substring(firstDot + 1)
+  if(thisRoute.handler === undefined || thisRoute.handler.findPath === undefined) {
```

Comment:
We'll need to make `findPath` a fully-documented public expectation for other router implementations. The idea behind the router currently is that anyone can implement various different ones and mix and match as needed. The docs, in addition to describing this new method, should describe how we would expect other router implementations to implement it as well.

We may need to consider what's going to happen if we just call `findPath` on someone's existing handler function that's out there as well, since we've never done that before, so don't know what we'd be breaking. Let's get some research going on this :)